packages/salesforcedx-vscode-core/src/commands/forceSourceDeploySourcePath.ts

```diff
@@ -51,79 +49,85 @@ export class LibraryDeploySourcePathExecutor extends DeployExecutor<
   }
 }
 
-export class LibraryPathsGatherer implements ParametersGatherer<string[]> {
-  private uris: vscode.Uri[];
-  public constructor(uris: vscode.Uri[]) {
-    this.uris = uris;
-  }
-  public async gather(): Promise<ContinueResponse<string[]>> {
-    const sourcePaths = this.uris.map(uri => uri.fsPath);
-    return {
-      type: 'CONTINUE',
-      data: sourcePaths
-    };
-  }
-}
-
-export async function forceSourceDeploySourcePath(sourceUri: vscode.Uri) {
+export const forceSourceDeploySourcePaths = async (
+  sourceUri: vscode.Uri | vscode.Uri[] |undefined,
+  uris: vscode.Uri[] | undefined
+) => {
   if (!sourceUri) {
-    const editor = vscode.window.activeTextEditor;
```

Comment:
@jag-sfdc I think what happened is that I originally had (pseudo code):
```
export async function forceSourceDeploySourcePath(sourceUri: vscode.Uri, uris: vscode.Uri[]) {
    if (uris) {
        forceSourceDeployMultipleSourcePaths()
        return;
    }

    // continue with the rest of forceSourceDeploySourcePath()...
```

When I consolidate and just used forceSourceDeployMultipleSourcePaths(), this check (lines 71-86) was removed & wasn't noticed, b/c 1) forceSourceDeployMultipleSourcePaths() didn't have this logic in it, and 2) the paths we exercised, never had an issue with that check, and we never ran into this issue.